qcodes/dataset/data_set.py

```diff
@@ -1247,15 +1255,25 @@ def _load_to_xarray_dataset(self, data: ParameterData) -> "xr.Dataset":
                 xrdataset.coords[str(dim)].attrs.update(paramspec_dict.items())
 
         xrdataset.attrs.update({
+            "name": self.name,
             "sample_name": self.sample_name,
             "exp_name": self.exp_name,
-            "snapshot": json.dumps(self.snapshot),
+            "snapshot": self.snapshot_raw or "null",
             "guid": self.guid,
             "run_timestamp": self.run_timestamp() or "",
             "completed_timestamp": self.completed_timestamp() or "",
-            "run_id": self.run_id
+            "captured_run_id": self.captured_run_id,
+            "captured_counter": self.captured_counter,
+            "run_id": self.run_id,
+            "run_description": serial.to_json_for_storage(self.description)
         })
 
```

Comment:
adding the raw, integer/float, represenation of timestamp. i think it's cleaner to NOT add an attibute if it's "None" other than add it as "null" (where a numeric value is expected) or as None-like value ("0" or "-1").
```suggestion
        if self.run_timestamp_raw is not None:
            xrdataset.attrs["run_timestamp_raw"]: self.run_timestamp_raw
        if self.completed_timestamp_raw is not None:
            xrdataset.attrs["completed_timestamp_raw"]: self.completed_timestamp_raw
```